File: euler/problem6.js
/*
The sum of the squares of the first ten natural numbers is,

12 + 22 + ... + 102 = 385
The square of the sum of the first ten natural numbers is,

(1 + 2 + ... + 10)2 = 552 = 3025
Hence the difference between the sum of the squares of the first ten natural 
numbers and the square of the sum is 3025 − 385 = 2640.

Find the difference between the sum of the squares of the first one hundred 
natural numbers and the square of the sum.
*/

function diffSquares(upperBound){
  var sumOfSquares = 0;
  var squareOfSum = 0;
  for(var i = 0; i <=upperBound; i++){
    sumOfSquares += (i * i);
    squareOfSum += i;
  }
  squareOfSum = Math.pow(squareOfSum, 2);
  return squareOfSum - sumOfSquares;
}

console.log(diffSquares(100));

File: euler/problem6.py
def diffOfSquares(upperBound):
  sumOfSquares = 0
  squareOfSums = 0
  for i in range(1, upperBound + 1):
    sumOfSquares += i ** 2
    squareOfSums += i
  squareOfSums = squareOfSums ** 2
  return squareOfSums - sumOfSquares


print diffOfSquares(100)